jdaviz/configs/mosviz/plugins/parsers.py

```diff
@@ -513,130 +517,190 @@ def _get_source_identifiers_by_hdu(hdus, filepaths=None, header_keys=['SOURCEID'
+    _warn_if_not_found(app, files_by_labels)
+
+    # parse relevant information from source catalog
+    cat_id_dict = {}
+    cat_file = files_by_labels[cat_key]
+    try:
+        cat_fields = ['label', 'sky_centroid.ra', 'sky_centroid.dec']
+        parsed_cat_fields = _fields_from_ecsv(cat_file, cat_fields,
+                                              delimiter=" ")
+    except KeyError:
+        # Older pipeline builds use different colname to distinguish sources
+        cat_fields[0] = 'id'
+        parsed_cat_fields = _fields_from_ecsv(cat_file, cat_fields,
+                                              delimiter=" ")
+    for row in parsed_cat_fields:
+        cat_id_dict[int(row[0])] = (row[1], row[2])
+
+    # set up a dictionary of datasets to add to glue
+    add_to_glue = {}
 
-    for image_file in file_lists["Direct Image"]:
-        im_split = image_file.split("/")[-1].split("_")
-        pupil = [x
-                 for x in image_file.split("/")[-1].split("_")
-                 if x[0] == "F" or x[0] == "f"][0]
+    # Parse images
+    image_dict = {}
 
-        image_label = "Image {} {}".format(im_split[0], pupil)
+    print("Loading: Images")
+    for image_file in files_by_labels["Direct Image"]:
+        # save label for table viewer
+        im_split = image_file.stem.split("_")[0]
+        pupil = fits.getheader(image_file, ext=0).get('PUPIL')
 
-        with fits.open(image_file) as file_obj:
-            data_iter = get_image_data_iterator(app, file_obj, "Image", ext=None)
-            data_obj = [d[0] for d in data_iter]  # We do not use the generated labels
-            image_data = data_obj[0]  # Grab the first one. TODO: Error if multiple found?
+        image_label = f"Image {im_split} {pupil}"
+        image_dict[pupil] = image_label
 
+        # save information from data file
         with fits.open(image_file) as temp:
-            filter_wcs[pupil] = temp[1].header
+            data_iter = get_image_data_iterator(app, temp, "Image", ext=None)
+            data_obj = [d[0] for d in data_iter]  # We do not use the generated labels
+            image_data = data_obj[0]  # Grab the first one.
+            # TODO: Error if multiple found?
 
         image_data.label = image_label
         add_to_glue[image_label] = image_data
 
-        image_dict[pupil] = image_label
-
-    # Parse 2D spectra
+    # initialize lists of data to be shown in table viewer
+    source_ids = []
+    ras = []
+    decs = []
+    image_add = []
+    spec_labels_1d = []
     spec_labels_2d = []
     filters = []
-    for f in ["2D Spectra C", "2D Spectra R"]:
 
-        for fname in file_lists[f]:
-            print(f"Loading: {f} sources")
+    # Parse 2D spectra
+    file_labels_2d = [k for k in files_by_labels.keys() if k.startswith("2D")]
+
+    for flabel in file_labels_2d:
+        for fname in files_by_labels[flabel]:
+            print(f"Loading: {flabel} sources")
```

Comment:
Is this print intentional? Or should we take advantage of debug snackbar logging?

Reply:
The print statements are grandfathered in and weren't my addition. Logging the process as cell output as opposed to a snackbar message has made debugging easier, but that may not be the best reason to continue the practice.

Reply:
OK, no prob. We can defer this.

Reply:
It was asked for by @camipacifici when the parser was first created.